gitbook-docs/data_model.md

```diff
@@ -171,9 +171,10 @@ A delta message can be recognised from the other types by the topmost level havi
 `updates` is the only required property.
 If `context` is missing it is assumed that the data is related to the `self` context.
 
-Context is a path from the root of the full tree. In this case 'vessels.urn:mrn:imo:mmsi:234567890'. All subsequent data is relative to
-that location. The context could be much more specific, e.g. 'vessels.urn:mrn:imo:mmsi:234567890.navigation', whatever is the common root
-of the updated data.
+Context is a path from the root of the full tree to the _container object_, which is always a vessel with the current Signal K version.
+The delimiter in the context path is `.` (period).
```

Comment:
So something that has been bugging me: why do we use `.` for the delimiter here but use the path delimiter `/` in the REST API? Can we be consistent?

Actually, this does not actually seem like a true path. When we say context "is always a vessel" then we are really specifying a context type (vessel) and identifier for a specific context (vesselId). Especially if we move away from the hierarchical model for values (#304) then it would be more flexible and easier to implement if we made `context` an object e.g.
```json
{
  "context": {
    "type": "vessel",
    "id": "urn:mrn:imo:mmsi:234567890"
  },
  "updates":[]
}
```
That means applications no longer need to parse the string to separate the two, and avoids any issues with needing to escape a `.` or `/` separator.